src/Adapter/News/NewsDataProvider.php

```diff
@@ -0,0 +1,175 @@
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\News;
+
+use Context;
+use PrestaShop\CircuitBreaker\Contract\CircuitBreakerInterface;
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use PrestaShop\PrestaShop\Adapter\Country\CountryDataProvider;
+use PrestaShop\PrestaShop\Adapter\Tools;
+use PrestaShop\PrestaShop\Adapter\Validate;
+
+/**
+ * Provide the news from https://www.prestashop.com/blog/
+ */
+class NewsDataProvider
+{
+    const NUM_ARTICLES = 2;
+
+    const CLOSED_ALLOWED_FAILURES = 3;
+    const CLOSED_TIMEOUT_SECONDS = 3;
+
+    const OPEN_ALLOWED_FAILURES = 3;
+    const OPEN_TIMEOUT_SECONDS = 3;
+    const OPEN_THRESHOLD_SECONDS = 86400; // 24 hours
+
+    /**
+     * @var CircuitBreakerInterface
+     */
+    private $circuitBreaker;
+
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @var int
+     */
+    private $contextMode;
+
+    /**
+     * @var CountryDataProvider
+     */
+    private $countryDataProvider;
+
+    /**
+     * @var Tools
+     */
+    private $tools;
+
+    /**
+     * @var Validate
+     */
+    private $validate;
+
+    /**
+     * NewsDataProvider constructor.
+     *
+     * @param CircuitBreakerInterface $circuitBreaker
+     * @param CountryDataProvider $countryDataProvider
+     * @param Tools $tools
+     * @param Configuration $configuration
+     * @param Validate $validate
+     * @param int $contextMode
+     */
+    public function __construct(
+        CircuitBreakerInterface $circuitBreaker,
+        CountryDataProvider $countryDataProvider,
+        Tools $tools,
+        Configuration $configuration,
+        Validate $validate,
+        $contextMode
+    ) {
+        $this->circuitBreaker = $circuitBreaker;
+        $this->configuration = $configuration;
+        $this->contextMode = $contextMode;
+        $this->countryDataProvider = $countryDataProvider;
+        $this->tools = $tools;
+        $this->validate = $validate;
+    }
+
+    /**
+     * @param string $isoCode
+     *
+     * @return array
+     *
+     * @throws \PrestaShopException
+     */
+    public function getData($isoCode)
+    {
+        $data = ['has_errors' => true, 'rss' => []];
+        $apiUrl = $this->configuration->get('_PS_API_URL_');
+
+        $blogXMLResponse = $this->circuitBreaker->call($apiUrl . '/rss/blog/blog-' . $isoCode . '.xml');
+
+        if (empty($blogXMLResponse)) {
+            $data['has_errors'] = false;
```

Comment:
Hi, this last modification seems weird. If no answer is received (nor cached) you should return `has_errors = true` no?
I get that when the connection or the service is not available the block must not be displayed. But then it should be managed in the front when an error is detected. Or the `has_errors` field is useless since it's always set to false.

Reply:
@jolelievre I just implement the rule :
> **What do we do while we don't get an answer?**
> Nothing - Currently if no answer the block displays a link to the PS blog "Find more news", we keep that.

For that, `$data['has_errors']` must be at `false` in the case where I don't get an answer (so `$blogXMLResponse = ''`). If there are XML errors, then `$data['has_errors'] = true`.

Reply:
Alright, perfect ;)